pkg/status/status.go

```diff
@@ -0,0 +1,93 @@
+package status
+
+import (
+	"context"
+	"time"
+
+	metallbv1alpha1 "github.com/metallb/metallb-operator/api/v1alpha1"
+	"github.com/pkg/errors"
+	appsv1 "k8s.io/api/apps/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+const (
+	ConditionAvailable   = "Available"
+	ConditionProgressing = "Progressing"
+	ConditionDegraded    = "Degraded"
+	ConditionUpgradeable = "Upgradeable"
+)
+
+func UpdateStatus(ctx context.Context, client k8sclient.Client, metallb *metallbv1alpha1.Metallb, condition string, reason string, message string) error {
+	metallb.Status.Conditions = getConditions(condition, reason, message)
+	if err := client.Status().Update(ctx, metallb); err != nil {
+		return errors.Wrapf(err, "could not update status for object %+v", metallb)
+	}
+	return nil
+}
+
+func getConditions(condition string, reason string, message string) []metav1.Condition {
+	conditions := getBaseConditions()
+	switch condition {
+	case ConditionAvailable:
+		conditions[0].Status = metav1.ConditionTrue
+		conditions[1].Status = metav1.ConditionTrue
+	case ConditionProgressing:
+		conditions[2].Status = metav1.ConditionTrue
+		conditions[2].Reason = reason
+		conditions[2].Message = message
+	case ConditionDegraded:
+		conditions[3].Status = metav1.ConditionTrue
+		conditions[3].Reason = reason
+		conditions[3].Message = message
+	}
+	return conditions
+}
+
+func getBaseConditions() []metav1.Condition {
+	now := time.Now()
+	return []metav1.Condition{
+		{
+			Type:               ConditionAvailable,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionUpgradeable,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionProgressing,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+		{
+			Type:               ConditionDegraded,
+			Status:             metav1.ConditionFalse,
+			LastTransitionTime: metav1.Time{Time: now},
+		},
+	}
+}
+
+func IsMetallbAvailable(ctx context.Context, client k8sclient.Client) (bool, error) {
+
+	ds := &appsv1.DaemonSet{}
+	err := client.Get(ctx, types.NamespacedName{Name: "speaker", Namespace: "metallb-system"}, ds)
```

Comment:
Use the namespace passed as parameter to the process, instead of hardcoding it

Reply:
done